Assets/Scripts/Enemy/EnemyHealth.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class EnemyHealth : MonoBehaviour
+{
+    public float startingHealth = 100;
+
+    private float currentHealth;
+
+    public float projectileDamage = 10f;
+
+    public Slider healthSlider;
+
+    public GameObject moneyPrefab;
+
+    private Animator animator;
+
+    private bool isDead;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = startingHealth;
+        animator = GetComponent<Animator>();
+
+        healthSlider.gameObject.SetActive(false);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (isDead) {
+            return;
+        }
+
+        if (CalculateDistanceFromPlayer() < 20)
+        {
+            healthSlider.gameObject.SetActive(true);
+        }
+        else
+        {
+            healthSlider.gameObject.SetActive(false);
+        }
+    }
+
+    void OnTriggerEnter(Collider other) {
+        if (isDead) {
+            return;
+        }
+        
+        if (other.CompareTag("Projectile")) {
+            TakeDamage(projectileDamage);
+        }
+    }
+
+    public void TakeDamage(float damage) {
+        currentHealth -= damage;
+        healthSlider.value = currentHealth;
+
+        if (currentHealth <= 0) {
+            Die();
+        }
+    }
+
+    void Die() {
+        isDead = true;
+        animator.SetBool("dieNow", true);
+        Destroy(healthSlider.gameObject);
+
+        GameObject money = Instantiate(moneyPrefab, transform.position, Quaternion.identity);
+        money.GetComponent<Rigidbody>().AddForce(Vector3.up * 5, ForceMode.Impulse);
```

Comment:
Why apply physics to it? Could we just have it appear over the dead zombie's body?

Reply:
it looked cool. i'll remove it though.